File: VB/TestMyTreeListForum/MyToolButtonViewInfo.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing



Namespace DevExpress.MyControl
    Public Class MyToolButtonViewInfo
        Private buttonField As MyToolButton
        Public ReadOnly Property Button() As MyToolButton
            Get
                Return buttonField
            End Get
        End Property



        Public Sub New(ByVal button As MyToolButton)
            Me.buttonField = button
            imageBoundsField = Rectangle.Empty
            captionBoundsField = Rectangle.Empty
            boundsField = Rectangle.Empty
        End Sub





        Private imageBoundsField As Rectangle
        Public ReadOnly Property ImageBounds() As Rectangle
            Get
                Return imageBoundsField
            End Get
        End Property




        Private captionBoundsField As Rectangle
        Public ReadOnly Property CaptionBounds() As Rectangle
            Get
                Return captionBoundsField
            End Get
        End Property


        Private boundsField As Rectangle
        Public ReadOnly Property Bounds() As Rectangle
            Get
                Return boundsField
            End Get
        End Property



        Public Sub CalcBounds(ByVal rect As Rectangle)
            boundsField = New Rectangle(rect.Location, rect.Size)
            If buttonField.ButtonType = MyButtonType.Link Then
                imageBoundsField = New Rectangle(New Point(boundsField.X, boundsField.Y + 2), New Size(10, 10))
                captionBoundsField = New Rectangle(imageBoundsField.Right + 2, boundsField.Y, boundsField.Right - imageBoundsField.Right, boundsField.Height)
                Return
            End If

            imageBoundsField = boundsField
        End Sub

        Private underCursorField As Boolean
        Public Property UnderCursor() As Boolean
            Get
                Return underCursorField
            End Get
            Set(ByVal value As Boolean)
                underCursorField = value
            End Set
        End Property
    End Class
End Namespace

File: CS/TestMyTreeListForum/MyToolButtonViewInfo.cs
using System;
using System.Collections.Generic;
using System.Drawing;



namespace DevExpress.MyControl
{
    public class MyToolButtonViewInfo
    {
        private MyToolButton button;
        public MyToolButton Button
        {
            get
            {
                return button;
            }
        }



        public MyToolButtonViewInfo(MyToolButton button)
        {
            this.button = button;
            imageBounds = Rectangle.Empty;
            captionBounds = Rectangle.Empty;
            bounds = Rectangle.Empty;
        }




        private Rectangle imageBounds;
        public Rectangle ImageBounds
        {
            get
            {
                return imageBounds;
            }
        }



        private Rectangle captionBounds;
        public Rectangle CaptionBounds
        {
            get
            {
                return captionBounds;
            }
        }



        private Rectangle bounds;
        public Rectangle Bounds
        {
            get
            {
                return bounds;
            }
        }



        public void CalcBounds(Rectangle rect)
        {
            bounds = new Rectangle(rect.Location, rect.Size);
            if (button.ButtonType == MyButtonType.Link)
            {
                imageBounds = new Rectangle(new Point(bounds.X, bounds.Y + 2), new Size(10, 10));
                captionBounds = new Rectangle(imageBounds.Right + 2, bounds.Y, bounds.Right - imageBounds.Right, bounds.Height);
                return;
            }

            imageBounds = bounds;
        }



        private bool underCursor;
        public bool UnderCursor
        {
            get
            {
                return underCursor;
            }
            set
            {
                underCursor = value;
            }
        }
    }
}
